parser.js

```diff
@@ -36,16 +36,24 @@ Parser.prototype._newPacket = function () {
 }
 
 Parser.prototype.parse = function (buf) {
+
   this._list.append(buf)
 
-  while ((this.packet.length != -1 || this._list.length > 0) &&
-         this[this._states[this._stateCounter]]()) {
-    this._stateCounter++
+  try {
+
+    while ((this.packet.length != -1 || this._list.length > 0) &&
+           (noError = this[this._states[this._stateCounter]]())) {
+      this._stateCounter++
 
-    if (this._stateCounter >= this._states.length) {
-      this._stateCounter = 0
+      if (this._stateCounter >= this._states.length) {
+        this._stateCounter = 0
+      }
     }
   }
+  catch (e) {
+
+    return this.emit('error', e);
+  }
```

Comment:
The problem is that _any_ try catch will make things slow. Moreover, any exception in node are better to crash the process anyway, as it is hard to know if you are not leaking memory/file descriptor/whatever.